### ISSUE TITLE
Nested Random Effects Model using MixedLM?

### ISSUE THREAD

**Issue by m-makarious** [statsmodels/statsmodels#8123]
#### Is your feature request related to a problem? Please describe
> A clear and concise description of what the problem is. Ex. I'm always frustrated when [...]

Wondering if there's a way using the `smf.mixedlm` function to specify different random effect formula for 2 variables of interest. 

Desired model: 
`Y ~ time * protein + time + [multiple FE covariates here...] + (1 | site) + (1 + time | subjectID)`

In the current implementation, I believe only this version of a model is possible to define (no random slopes, just random intercepts): 
`Y ~ time * protein + time + [multiple FE covariates here...] + (1 | site) + (1 | subjectID)`

by defining the dictionary passed through `vc_formula` as the following 
```
vcf = {"site": "0 + C(site)", "IID": "0 + C(subjectID)"}
```

#### Describe the solution you'd like
> A clear and concise description of what you want to happen.

Looking for a way to have different random effect formulas depending on the variable, so the model can be defined as desired above.

#### Describe alternatives you have considered
> A clear and concise description of any alternative solutions or features you have considered.

From my understanding, statsmodel implements patsy (and therefore must be maintaining some version of patsy for the mixedlm function). This issue was opened on patsy here, https://github.com/pydata/patsy/issues/130 though looks as though patsy is no longer maintained and a PR has not been opened to tackle this. Outside of Python, will be considering doing this in R if not possible.

#### Additional context
> Add any other context about the feature request here.

If this is already possible, I'd love to know! Feel free to point me in the right direction and close this FR :) just running out of luck on Google

**Comment by kshedden** [comment]
I don't think you can get exactly the model that you want, but you can get one step closer than where you are now, with a model that is equivalent to this one:

Y ~ time * protein + time + [multiple FE covariates here...] + (1 | site) + (1 | subjectID) + (0 + time | subjectID)

Using this:

vcf = {"site": "0 + C(site)", "IID1": "0 + C(subjectID)", "IID2": "0 + C(subjectID):time"}

The only difference here is that the random slope and random intercept (for subjects) are correlated whereas you may want them to be correlated.  Perhaps by centering time, either globally or within subjects, there may be less need for a correlation.

Note in this approach that you need to set the groups variable to be the same for all observations.
